jetbrains-core/src/software/aws/toolkits/jetbrains/services/dynamic/DynamicResourcesUpdateManager.kt

```diff
@@ -128,29 +129,19 @@ internal class DynamicResourceUpdateManager(private val project: Project) {
 
     private fun getProgress() {
         var size = pendingMutations.size
-
+        var isValidToken = true
         while (size > 0) {
             val mutation = pendingMutations.remove()
             if (!mutation.status.isTerminal()) {
                 val client = mutation.connectionSettings.awsClient<CloudControlClient>()
                 val progress = try {
                     client.getResourceRequestStatus { it.requestToken(mutation.token) }
                 } catch (e: Exception) {
```

Comment:
Instead of multiple catch blocks you can have a single-top-level catch block and then use `when` to differentiate them:

```kt
} catch (e: Exception) {
  val shouldRequeue = when(e) {
    is RequestTokenNotFoundException -> {
      unableToGetProgress(e, mutation)
      false
    }
    else -> true
}
```

Reply:
yep, done